### PR TITLE
Add hexagon icon shape

### PR DESCRIPTION
**Pull request by SelfRef**
## Description

This PR adds a new icon shape - hexagon. Basically this is port of my own PR for NeoApplications/Neo-Launcher#388

From my tests, it works well:

Settings|Home|List
---|---|---
Screenshot_20240822_192628 | Screenshot_20240822_014056 | Screenshot_20240822_014119

## Type of change

:x: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:white_check_mark: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by tgex0** [comment]
Not a big deal, but currently, all icon shapes are listed in the UI alphabetically (except for System at the very top). Slotting Hexagon between Octagon and One UI would break this trend.

File: lawnchair/res/values-pl-rPL/strings.xml [hunk 1]
```diff
@@ -160,6 +160,7 @@
     <string name="icon_shape_egg">Jajko</string>
     <string name="icon_shape_cupertino">iOS</string>
     <string name="icon_shape_octagon">Ośmiokątny</string>
+    <string name="icon_shape_hexagon">Sześciokątny</string>
```
Comment (validcube) [inline lawnchair/res/values-pl-rPL/strings.xml]: Lawnchair Launcher only uploads the source text (strings.xml) to Crowdin so this change is likely going to be reverted in the next translation sync. 

I'd recommend translating this string once this PR has been merged (and Crowdin Upload CI has already done its job.)

**Comment by SelfRef** [comment]
> Not a big deal, but currently, all icon shapes are listed in the UI alphabetically (except for System at the very top). Slotting Hexagon between Octagon and One UI would break this trend.

I saw the order is hard-coded, so even if for English it may work like that, for other languages it won't, isn't it?

But yeah, I changed the order so it fits the current scheme.

File: lawnchair/res/values-pl-rPL/strings.xml (hunk 1 above)
Comment (SelfRef) [reply lawnchair/res/values-pl-rPL/strings.xml]: Sure, I removed the translation from PR.

**Review by MrSluffy — APPROVED** [review]

**Review by SuperDragonXD — APPROVED** [review]

**Comment by RustoMCSpit** [comment]
is it possible to have this icon but with the top and bottom lines parallel to the x axis. as in, just rotate it a bit

**Comment by SelfRef** [comment]
> is it possible to have this icon but with the top and bottom lines parallel to the x axis. as in, just rotate it a bit

I thought about that position at first, but it doesn't look good because most of the icons have vertical sides and often a sticking part at the top or bottom.

Maybe someone will be willing to modify my PR to make a rotated version. I don't want to set up Android Studio again, honestly.

**Pull request merged**